src/targets.jl

```diff
@@ -1,31 +1,31 @@
 abstract type Target end
 
 # For finding eigenvalues with the smallest magnitude
-struct SM{T} <: Target end
+struct SM <: Target end
 
 # For finding eigenvalues with the largest magnitude
-struct LM{T} <: Target end
+struct LM <: Target end
 
 # For finding eigenvalues with the largest real part
-struct LR{T} <: Target end
+struct LR <: Target end
 
 # For finding eigenvalues with the smallest real part
-struct SR{T} <: Target end
+struct SR <: Target end
 
 # For finding eigenvalues with the largest imaginary part
-struct LI{T} <: Target end
+struct LI <: Target end
 
 # For finding eigenvalues with the smallest imaginary part
-struct SI{T} <: Target end
+struct SI <: Target end
 
-sort_vals!(λs,::Type{LM}) = sort!(λs, by=abs, rev=true)
+sort_vals!(λs,::LM) = sort!(λs, by=abs, rev=true)
 
-sort_vals!(λs,::Type{SM}) = sort!(λs, by=abs)
+sort_vals!(λs,::SM) = sort!(λs, by=abs)
 
-sort_vals!(λs,::Type{LR}) = sort!(λs, lt = f(a,b) = abs(real(a)) > abs(real(b)))
+sort_vals!(λs,::LR) = sort!(λs, by = real)
 
-sort_vals!(λs,::Type{SR}) = sort!(λs, lt = f(a,b) = abs(real(a)) < abs(real(b)))
+sort_vals!(λs,::SR) = sort!(λs, by = f(x) = - real(x))
```

Comment:
Either use closures `x -> -real(x)` or use `by = real, rev = true`; doesn't make sense to give this function a name.